packages/features/auth/signup/handlers/registerUnverifiedMentorHandler.ts

```diff
@@ -13,11 +12,15 @@ import { prefillAvatar } from "../utils/prefillAvatar";
 
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
   const data = req.body;
-  const { mentorName, email, password, language, token } = registerMentorSchema.parse(data);
+  const { mentorName, email, password, language, key } = registerMentorSchema.parse(data);
+
+  if (key != process.env.USER_CREATION_ACCESS_KEY) {
+    return res.status(403).json({ message: "Not allowed to create user" });
+  }
 
   const userEmail = email.toLowerCase();
   const username = userEmail.split("@")[0];
-  let hashedPassword;
+  const hashedPassword = password ? await hashPassword(password) : await hashPassword("");

```

Comment:
for increased security, instead of hashPassword(""), put something like 

import * as crypto from "node:crypto";
const id = crypto.randomBytes(20).toString('hex');